### DISCUSSION TITLE
TS2339: Property 'id' does not exist on type 'Record"key", string>'.

### DISCUSSION THREAD

**Discussion by ribeaud** [react-hook-form/react-hook-form#7754] [Q&A] [answered]
Hi. We invoking `useFieldArray` as following:
```ts
const { fields, append, remove } = useFieldArray({
    control,
    name: 'users',
    keyName: 'key',
  });
```
**User** is a pure DB entity, this is why we're using _keyName_. _id_ is already taken and we need it on frontend side.
A bit further, we try to access this value with:
```ts
!fields.some((field) => {
        // Here getting a TS exception: TS2339: Property 'id' does not exist on type 'Record"key", string>'.
        const fieldId = field.id;
        return ...
      })
```
What should I do to properly get rid of the **TS** error? I tried out different strategies (like extending `FieldValues` type with `& {id: string}`) but none worked out.

Any help would be appreciated here. Many thanks.

**Comment by bluebill1049** [comment, marked as answer]
which version are you using? change `keyName` is no longer required.

**Reply by ribeaud** [reply]
Thanks. But now I am a bit puzzled... 🤔 

We are using _7.25.3_ (so kind of the latest). And, AFAIK, _keyName_ was needed to avoid the overwriting of _id_ with an **uuid** value (see https://github.com/react-hook-form/react-hook-form/issues/7562).

**Reply by Moshyfawn** [reply]
You'll see the IDs being overridden with RHF when mapping through the fields, but `onSubmit` you'll get your data set with your initial IDs instead

**Reply by ribeaud** [reply]
But then, we must be doing something wrong because I saw those **uuids** set as _id_ on _backend_ side, after submission if we do NOT use `keyName`. It used to work with _7.22.5_ without `keyName` but stops working afterwards and this is the reason we started to use `keyName`.

Is there anything I have to do to make `onSubmit` working the way you're describing? Or is there anything I should NOT do to make `onSubmit` working properly?

**Reply by Moshyfawn** [reply]
There shouldn't be any additional step to get the initial IDs in your submit handler, tbh. I couldn't reproduce the behavior you've described with RHF ver. 7.25.3 in [this CSB](https://codesandbox.io/s/cool-pike-p8kt3?file=/src/App.tsx). Can you take a look and potentially tweak the code to closer match your use-case?

**Reply by ribeaud** [reply]
Thanks guys. Owing to the example, I was able to fix the situation on our side. You rock! 💪
